x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/resources_stats_collector.ts

```diff
@@ -0,0 +1,166 @@
+ * 2.0.
+ */
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { Logger } from '@kbn/core/server';
+import { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import { LATEST_FINDINGS_INDEX_DEFAULT_NS } from '../../../../common/constants';
+import type { CspmResourcesStats } from './types';
+
+interface ResourcesStats {
+  accounts: {
+    buckets: AccountEntity[];
+  };
+}
+
+export interface AccountEntity {
+  key: string; // account_id
+  doc_count: number;
+  resource_type: {
+    buckets: ResourceType[];
+  };
+}
+
+interface ResourceType {
+  key: string; // resource_type
+  doc_count: number;
+  resource_sub_type: {
+    buckets: ResourceSubType[];
+  };
+}
+
+interface ResourceSubType {
+  key: string; // resource_sub_type
+  doc_count: number;
+  evaluation: {
+    buckets: EvaluationStats[];
+  };
+}
+
+interface EvaluationStats {
+  key: string; // passed / failed
+  doc_count: number;
+}
+
+export const getResourcesStats = async (
+  esClient: ElasticsearchClient,
+  logger: Logger
+): Promise<CspmResourcesStats[]> => {
+  try {
+    const isIndexExists = await esClient.indices.exists({
+      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+    });
+
+    if (isIndexExists) {
+      return await getResourceStats(esClient, LATEST_FINDINGS_INDEX_DEFAULT_NS);
+    }
+
+    return [];
+  } catch (e) {
+    logger.error(`Failed to get resources stats ${e}`);
+    return [];
+  }
+};
+
+const getResourceStats = async (
+  esClient: ElasticsearchClient,
+  index: string
+): Promise<CspmResourcesStats[]> => {
+  const resourcesStatsResponse = await esClient.search<unknown, ResourcesStats>(
+    getResourcesStatsQuery(index)
+  );
+  const resourcesStats = resourcesStatsResponse.aggregations
+    ? parseResourcesStatsQueryResponse(resourcesStatsResponse.aggregations)
+    : [];
+  return resourcesStats;
+};
+
+const parseResourcesStatsQueryResponse = (resourcesStats: ResourcesStats): CspmResourcesStats[] => {
+  const accounts = resourcesStats.accounts.buckets;
+
+  const parsedResourcesStats = accounts.map((account) => {
+    const accountId = account.key;
+    return account.resource_type.buckets.map((resourceType) => {
+      return resourceType.resource_sub_type.buckets.map((resourceSubType) => {
+        const evaluation = getEvaluationStats(resourceSubType);
+        return {
+          account_id: accountId,
+          resource_type: resourceType.key,
+          resource_type_doc_count: resourceType.doc_count,
+          resource_sub_type: resourceSubType.key,
+          resource_sub_type_doc_count: resourceSubType.doc_count,
+          ...evaluation,
+        };
+      });
+    });
+  });
+  return parsedResourcesStats.flat(2);
+};
+
+const getEvaluationStats = (resourceSubType: ResourceSubType) => {
+  const passed =
+    resourceSubType.evaluation.buckets.find((evaluation) => evaluation.key === 'passed')
+      ?.doc_count ?? 0;
```

Comment:
minimal impact but using `||` would be better here. we want to return `0` in any [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) case, and not just `null` or `undefined`.

![image](https://user-images.githubusercontent.com/51442161/208966174-1437059b-1bd0-4905-8981-8ecef700ea9a.png)